dotCMS/src/main/java/com/dotcms/publishing/PublisherAPIImpl.java

```diff
@@ -1,40 +1,61 @@
-import com.dotcms.publishing.manifest.ManifestBuilder;
-import java.io.File;
-import java.io.IOException;
-import java.util.ArrayList;
-import java.util.Date;
-import java.util.List;
-import java.util.Map;
-import java.util.concurrent.ConcurrentHashMap;
 import com.dotcms.business.CloseDBIfOpened;
 import com.dotcms.publisher.business.PublishAuditAPI;
 import com.dotcms.publisher.business.PublishAuditHistory;
 import com.dotcms.publisher.business.PublishAuditStatus;
+import com.dotcms.publishing.manifest.CSVManifestBuilder;
+import com.dotcms.publishing.manifest.ManifestBuilder;
 import com.dotcms.publishing.output.BundleOutput;
 import com.dotcms.system.event.local.business.LocalSystemEventsAPI;
 import com.dotcms.system.event.local.type.pushpublish.PushPublishEndEvent;
 import com.dotcms.system.event.local.type.pushpublish.PushPublishStartEvent;
 import com.dotcms.system.event.local.type.staticpublish.StaticPublishEndEvent;
 import com.dotcms.system.event.local.type.staticpublish.StaticPublishStartEvent;
+import com.dotcms.util.YamlUtil;
 import com.dotmarketing.business.APILocator;
 import com.dotmarketing.business.Role;
 import com.dotmarketing.exception.DotDataException;
 import com.dotmarketing.exception.DotSecurityException;
+import com.dotmarketing.util.Config;
 import com.dotmarketing.util.Logger;
 import com.dotmarketing.util.PushPublishLogger;
 import com.dotmarketing.util.UtilMethods;
-import com.google.common.annotations.VisibleForTesting;
 import com.liferay.portal.model.User;
+import org.apache.commons.io.FileUtils;
 
+import java.io.File;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.Date;
+import java.util.List;
+import java.util.concurrent.CopyOnWriteArrayList;
+import java.util.stream.Stream;
+
+import static com.dotcms.content.elasticsearch.constants.ESMappingConstants.MOD_DATE;
+
+/**
+ * Implementation class for the {@link PublisherAPI}.
+ * <p>Every specific Bundle {@link Publisher} has its own specific list of data Bundlers -- {@link IBundler} -- which
+ * determine what and how specific pieces of information are added to a given Bundle. A MANIFEST file is added to all
+ * bundles in order to indicate why specific data was included or excluded in them.</p>
+ *
+ * @author Jason Tesser
+ * @since Mar 23rd, 2012
+ */
 public class PublisherAPIImpl implements PublisherAPI {
 
     private final PublishAuditAPI publishAuditAPI = PublishAuditAPI.getInstance();
     private final LocalSystemEventsAPI localSystemEventsAPI = APILocator.getLocalSystemEventsAPI();
-    private final Map<String,FilterDescriptor> loadedFilters = new ConcurrentHashMap<>();
-
+    private List<FilterDescriptor> filterList = new CopyOnWriteArrayList<>();
+    /** Path where the YAML files are stored */
+    private final String PUBLISHING_FILTERS_FOLDER =
```

Comment:
I am not sure if this should be a lazy or not

Reply:
Done!